### PR TITLE
Fix for Bug "Incorrect behaviour when there is "describe" block inside another "describe' block"

### PR DESCRIPTION
**Pull request by rpgeeganage**
With this PR,
1. Try to fix the bug https://github.com/fatso83/mini-mocha/issues/8
In order to fix the bug, I added functionality to process each nested `describe` block in the corresponding parent `describe` block.
2. Some formatting.

Looking forward for feedbacks.

### CODE REVIEW

File: lib/execute.js [hunk 1]
```diff
@@ -1,69 +1,129 @@
...
-  for (const ic of describeBlock.itCollection) {
-    await executeItFlow(describeBlock, ic);
-  }
-
-  // After
-  if (describeBlock.afterHook) {
-    await executeHook("after", describeBlock.afterHook);
-  }
-}
+    console.log(describeBlock.title);
```
Comment (fatso83) [inline lib/execute.js]: why `console.log` here and `log` elsewhere?

Comment (rpgeeganage) [reply lib/execute.js]: I'll move it to log. sorry about that.

File: lib/queue.js [hunk 2]
```diff
@@ -16,43 +15,31 @@ const queueHandler = new events.EventEmitter();
...
+    const exec = type === taskType.it ? executeIt(title, fn) : executeFn(fn).then(() => executeDescribe(fn[metaData]));
 
-  exec
-    .then(() => (isProcessing = false))
-    .then(() => processQueue())
-    .catch(() => process.exit(1));
+    exec.then(() => (isProcessing = false))
+        .then(() => processQueue())
+        .catch(error => {
+            console.log(error);
```
Comment (fatso83) [inline lib/queue.js]: console.error

File: lib/execute.js [hunk 3]
```diff
@@ -1,69 +1,129 @@
...
 async function executeHook(eachHookTitle, hook) {
-  return executeIt(eachHookTitle, hook, true);
+    return executeIt(eachHookTitle, hook, true);
+}
+
+// This is to handle nested Describe blocks recursively
+async function innerExecutor(otherDescribe, beforeEach = [], afterEach = [], itBlocks = [], afterHooks = []) {
+    await executeFn(otherDescribe);
+    const metaDataSection = otherDescribe[metaData];
+    console.log(metaDataSection.title);
```
Comment (fatso83) [inline lib/execute.js]: same

**Comment by rpgeeganage** [comment]
@fatso83 ,
I have changed the test case. Can you please look at the output and see, whether it is correct?
Thanks.

**Comment by rpgeeganage** [comment]
I fixed the describe processing and output should be fine now 😄

**Pull request merged**